include/runtime.h

```diff
@@ -118,8 +129,7 @@ namespace detail {
 		runtime(const runtime&) = delete;
 		runtime(runtime&&) = delete;
 
-		void handle_buffer_registered(buffer_id bid);
-		void handle_buffer_unregistered(buffer_id bid);
+		void register_buffer(buffer_id bid, const range<3>& range, bool host_initialized);
```

Comment:
Presumably this function is required because of forward declarations? Maybe you could add a comment.

Reply:
It's so I don't need to include `task_manager.h` and `distributed_graph_generator.h` here. Comment added.